spyder/plugins/pylint/tests/test_pylint.py

```diff
@@ -259,5 +337,35 @@ def test_pylint_max_history_conf(pylint_plugin, pylint_test_scripts):
     assert 'test_script_2.py' in pylint_widget.curr_filenames[0]
 
 
+def test_pylint_ignore_issue_in_a_function_definition(
+        ignore_rules_test_helper, qtbot, pylint_plugin_with_real_editor):
+    ignore_rules_test_helper.add_function_with_bad_name()
+    pylint_widget = pylint_plugin_with_real_editor.get_widget()
+    pylint_widget.set_filename(ignore_rules_test_helper.python_file_path)
+
+    pylint_plugin_with_real_editor.start_code_analysis(
+        filename=ignore_rules_test_helper.python_file_path)
+    wait_until_code_analysis_finish(pylint_widget, qtbot)
+
+    ignored_rule = ignore_rules_test_helper.\
+        ignore_first_item_that_matches_with(
+            ignore_rules_test_helper.bad_function_name, pylint_widget
+        )
+
+    wait_until_code_analysis_finish(pylint_widget, qtbot)
+
+    with open(ignore_rules_test_helper.python_file_path, mode="r") as file:
```

Comment:
```suggestion
    with open(ignore_rules_test_helper.python_file_path, mode="r",
              encoding="utf-8") as file:
```

For correctness, since you wrote the file with an explicitly specified encoding (the default is OS- and user-dependent, not UTF-8, and might not even be ASCII-compatible).